src/Std/Data/DHashMap/Internal/RawLemmas.lean

```diff
@@ -2757,6 +2776,357 @@ theorem get!_union_of_contains_eq_false_right [EquivBEq α] [LawfulHashable α]
+    (h₂ : m₂.val.WF) {k : α} (h : m₁.contains k = false) :
+    (m₁.inter m₂).getKey! k = default := by
+  revert h
+  simp_to_model [inter, getKey!, contains] using List.getKeyD_filter_containsKey_of_not_containsKey_left
+
+/- getEntry? -/
+theorem getEntry?_inter [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF) {k : α} :
+    (m₁.inter m₂).getEntry? k =
+    if m₂.contains k then m₁.getEntry? k else none := by
+  simp_to_model [inter, getEntry?, contains] using List.getEntry?_filter_containsKey
+
+theorem getEntry?_inter_of_contains_right [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (h : m₂.contains k) :
+    (m₁.inter m₂).getEntry? k = m₁.getEntry? k := by
+  revert h
+  simp_to_model [inter, getEntry?, contains] using List.getEntry?_filter_containsKey_of_containsKey_right
+
+theorem getEntry?_inter_of_not_contains_left [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (h : m₁.contains k = false) :
+    (m₁.inter m₂).getEntry? k = none := by
+  revert h
+  simp_to_model [inter, getEntry?, contains] using List.getEntry?_filter_containsKey_of_not_containsKey_left
+
+theorem getEntry?_inter_of_not_contains_right [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} (h : m₂.contains k = false) :
+    (m₁.inter m₂).getEntry? k = none := by
+  revert h
+  simp_to_model [inter, getEntry?, contains] using List.getEntry?_filter_containsKey_of_not_containsKey_right
+
+/- getEntry -/
+theorem getEntry_inter [EquivBEq α] [LawfulHashable α] (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    {k : α} {h_contains : (m₁.inter m₂).contains k} :
+    (m₁.inter m₂).getEntry k h_contains =
+    m₁.getEntry k (((@contains_inter_iff α β _ _ m₁ m₂ _ _ h₁ h₂ k).1 h_contains).1) := by
+  simp_to_model [inter, getEntry, contains] using List.getEntry_filter_containsKey
+
+/- size -/
+theorem size_inter_le_size_left [EquivBEq α] [LawfulHashable α]
+    (h₁ : m₁.val.WF) (h₂ : m₂.val.WF) :
+    (m₁.inter m₂).1.size ≤ m₁.1.size := by
+  simp_to_model [inter, size] using List.length_filter_le
+
+theorem size_inter_le_size_right [EquivBEq α] [LawfulHashable α]
+    (h₁ : m₁.val.WF) (h₂ : m₂.val.WF) :
+    (m₁.inter m₂).1.size ≤ m₂.1.size := by
+  simp_to_model [inter, size, contains] using List.length_filter_containsKey_le
+
+theorem size_inter_eq_size_left [EquivBEq α] [LawfulHashable α]
+    (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    (h : ∀ (a : α), m₁.contains a → m₂.contains a) :
+    (m₁.inter m₂).1.size = m₁.1.size := by
+  revert h
+  simp_to_model [inter, size, contains] using List.length_filter_containsKey_eq_length_left
+
+theorem size_inter_eq_size_right [EquivBEq α] [LawfulHashable α]
+    (h₁ : m₁.val.WF) (h₂ : m₂.val.WF)
+    (h : ∀ (a : α), m₂.contains a → m₁.contains a) :
+    (m₁.inter m₂).1.size = m₂.1.size := by
+  revert h
+  simp_to_model [inter, size, contains] using List.length_filter_containsKey_of_length_right
+
+
+/- isEmpty -/
```

Comment:
There is also the lemma `(m1.inter m2).isEmpty <-> \fo k, m1.contains k -> m2.contains k = false`.